CONTRIBUTING.md

```diff
@@ -0,0 +1,143 @@
+For example, start by explaining how You configured Gateway, e.g. which command exactly You used in the terminal, or how You configured network-related Linux services. When listing steps, **don't just say what You did but explain how You did it**. For example, if You changed Gateway configuration, explain what exactly You have changed in comparison with previously working case, also, specify what You are trying to achieve, maybe there could be suggested confidently working scenario.
+* **Describe the behaviour You observed after following the steps** and point out what exactly is the problem with that behaviour.
+* **Explain which behaviour You expected to see instead and why.**
+* **If makes sense, include screenshots and animated GIFs** which show You following the described steps and clearly demonstrate the problem.
+
+Provide more context by answering these questions:
+* **Did the problem start happening recently** (e.g. after updating to a new version of Gateway) or was this always a problem?
+* If the problem started happening recently, **can You reproduce the problem in an older version of Gateway?** What's the most recent version in which the problem doesn't happen?
+
+
+* **Can You reliably reproduce the issue?** If not, provide details about how often the problem happens and under which conditions it normally happens.
+
+Include details about Your configuration and environment:
+
+* **Which released version are You using?** You could find it in a special file.
+* **What is the name and version of the OS You are using**?
+
+### Suggesting Enhancements
+
+Before creating enhancement suggestions, please check the repository issue list as You might find out that You don't need to create one. When You are creating an enhancement suggestion, please include as many details as possible. Fill in [the template](TEMPLATES/feature_request.md), including the steps that You imagine You would take if the feature You're requesting existed.
+Explain the feature and include additional details:
+
+* **Use a clear and descriptive title** for the issue to quickly figure out the idea of suggestion.
+* **Provide a step-by-step description of the suggested enhancement** in as many details as possible.
+* **Describe the current behaviour** and **explain which behaviour You expected to see instead** and why.
+* **Explain why this enhancement would be useful**
+* **List some other applications where this enhancement exists.**
+* **Specify which version of Gateway You're using.** You can get a version from special config file
+* **Specify the name and version of the OS You are using.**
+
+### Pull Requests
+
+Here the checklist of what You need to create a successful pull request:
+1. Point issue on the repository
+2. Descriptive commit messages. [Checkout the guide](#git-commit-messages)
+3. In case You have changed the behaviour of some feature/ widely used function, etc....
```

Comment:
Contributors must change a documentation file located in each directory if a feature introduces considerable functionality, new concepts or design changes. We need to define the approach to follow regarding the documentation file in each code directory.

Reply:
I mentioned these 2 lines under, but I would make it more descriptive.
But, I think your comment would be more applicable to the "Documentation maintenance" chapter.

Reply:
In any case, the "Documentation Maintenance" chapter, is all about approach.
I believe we should apply one approach to all documentation files.